File: src/app/modules/m1-school-office/data-access/models/school-shift.ts
export interface SchoolShift {
  id: number;
  schoolId: number;
  shiftNameAr: string;
  shiftNameEn: string;
  startTime: string;
  endTime: string;
  shiftCode: string | null;
  totalPeriodsCount: number;
  periodDurationMinutes: number;
  breakDurationMinutes: number;
  isActive: boolean;
}

export type CreateSchoolShiftDto = Omit<SchoolShift, 'id' | 'isActive'>;

export type UpdateSchoolShiftDto = Omit<SchoolShift, 'schoolId' | 'isActive'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolShift.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class SchoolShift : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string ShiftNameAr { get; set; } = string.Empty;
    public string ShiftNameEn { get; set; } = string.Empty;
    public string StartTime { get; set; } = string.Empty; // e.g. "07:00"
    public string EndTime { get; set; } = string.Empty; // e.g. "13:30"
    public string? ShiftCode { get; set; }
    public int TotalPeriodsCount { get; set; } = 7;
    public int PeriodDurationMinutes { get; set; } = 45;
    public int BreakDurationMinutes { get; set; } = 30;
    public bool IsActive { get; set; } = true;

    // Navigation Property
    public virtual School? School { get; set; }
}
